GAE/src/org/akvo/flow/events/EventLogger.java

```diff
@@ -129,14 +132,24 @@ private void notifyLog() {
 
     private void storeEvent(Map<String, Object> event, Date timestamp) {
         try {
+            DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
             ObjectMapper m = new ObjectMapper();
             StringWriter w = new StringWriter();
             m.writeValue(w, event);
-            BaseDAO<EventQueue> eventDao = new BaseDAO<EventQueue>(EventQueue.class);
-            EventQueue eventQ = new EventQueue();
-            eventQ.setCreatedDateTime(timestamp);
-            eventQ.setPayload(w.toString());
-            eventDao.save(eventQ);
+
+            Entity entity = new Entity("EventQueue");
+            entity.setProperty("createdDateTime", timestamp);
+            entity.setProperty("lastUpdateDateTime", timestamp);
+
+            String payload = w.toString();
+            if (payload.length() > Constants.MAX_LENGTH) {
+                entity.setProperty("payloadText", new Text(payload));
+            } else {
+                entity.setProperty("payload", w.toString());
```

Comment:
we're calling `w.toString()` 2 times, `payload` local variable already has the String value, apologies for not catching it earlier